src/parser/parser.ts

```diff
@@ -181,7 +180,7 @@ export class Parser {
             }
         }
         if (type in tokens) {
-            if (tok.type === tokens[type]) {
+            if (tok.type === tokens[type as keyof typeof tokens]) {
```

Comment:
🤯

Reply:
Yeah. This is the same issue as the `Map.has` code. 

typescript doesn't let you use arbitrary strings to look up certain objects. And enum objects seem to be one of them.